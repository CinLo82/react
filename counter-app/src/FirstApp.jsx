import React from 'react'
import PropTypes from 'prop-types'

export const FirstApp = ( { title, subtitle, name }) => {

	return (
		<>
			<h1 data-testid="test-title">{ title }</h1>
			<h2>{subtitle}</h2>
            <h2>{subtitle}</h2>
            <p>{ name }</p>
			 
		</>
	)
}

FirstApp.propTypes = {
	title: PropTypes.string.isRequired,
    subtitle: PropTypes.string,
}

FirstApp.defaultProps = {
    name: 'Cinlo Losada',
    subtitle: 'No hay subtitulo',

}
