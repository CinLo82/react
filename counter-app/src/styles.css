body{
    background-color: #0b3e70;
    color: white;
    font-family: Arial, sans-serif;
}

button{
    background-color: #f0ad4e;
    font-size: larger;
    color: white;
    border: none;
    padding: 10px 20px;
    margin: 10px;
    border-radius: 5px;
    cursor: pointer;
}