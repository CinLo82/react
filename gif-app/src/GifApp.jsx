import { useState } from 'react'
import { AddCategory } from './components/AddCategory'

export const GifApp = () => {

    const [categories, setCategories] = useState(['One Punch', 'Dragon Ball', 'Naruto'])

    const onAddCategory = ( newCategory ) => {
        console.log(newCategory)
        setCategories([newCategory, ...categories])
    }


    return (
        <>
            <h1>GifApp</h1>
            <AddCategory
                //setCategories={setCategories}
                onNewCategory={ onAddCategory }
      
            />

            <ol>
                {
                    categories.map( category => {
                        return (
                        <li key={category}>{category}
                        </li>
                        )
                    })
                }
            </ol>
        </>
    )
}
