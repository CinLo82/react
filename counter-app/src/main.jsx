import React from 'react'
import ReactDOM from 'react-dom/client'
import './styles.css'
 import { CounterApp } from './CounterApp'
{/*import { FirstApp } from './FirstApp'*/}


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
         <CounterApp value={100} />
       {/*  <FirstApp title='hola, soy cinlo' subtitle='este es un subtitulo' /> */}
    </React.StrictMode>
)

