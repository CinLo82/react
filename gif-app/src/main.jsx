import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
import { GifApp } from '../GifApp.jsx'

createRoot(document.getElementById('root')).render(
  <StrictMode>
    <GifApp />
  </StrictMode>,
)
