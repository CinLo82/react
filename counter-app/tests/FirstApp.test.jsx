import { getAllByAltText, render } from "@testing-library/react" 
import { FirstApp } from "../src/FirstApp"

describe('Pruebas en <FirstApp />', () => {
  
    /*
    test('debe hacer match con el snapshot', () => {

            const title = 'Hola, soy Goku';
            render( <FirstApp title={title} /> )
            const { container } = render( <FirstApp title={title} /> );
            expect( container ).toMatchSnapshot()
        })
    */
   

    test('debe de mostrar el titulo en un h1', () => {
        const title = 'Hola, soy Goku';
        const { container, getByText, getByTestId } = render( <FirstApp title={title} /> );

        expect( getByText(title) ).toBeTruthy()

        // const h1 = container.querySelector('h1');
        // expect(h1.innerHTML).toContain(title) // Contiene el texto

        // expect( getByTestId('test-title') ).toBeTruthy() // si existe
        expect( getByTestId('test-title').innerHTML ).toContain(title) 
    })

    test('debe de mostrar el subtitulo enviado por props', () => {
        const title = 'Hola, soy Goku';
        const subtitle = 'Soy un subtitulo';
        const { container, getByText, getAllByText } = render( 
            <FirstApp 
                title={title} 
                subtitle={subtitle} 
            /> 
        );

        expect( getByText(subtitle) ).toBeTruthy() 
        // getByText es una funcion que busca un texto en el componente
        // getAllByText es una funcion que busca (mas de una vez) los textos en el componente, regresa un arreglo

        //expect(getAllByText(subtitle).length).toBe(2)
    })
})
