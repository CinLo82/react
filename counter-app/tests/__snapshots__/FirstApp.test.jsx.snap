// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, soy Goku
  </h1>
  <h2>
    No hay subtitulo
  </h2>
  <p>
    Cinlo Losada
  </p>
</div>
`;
